client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -218,7 +307,12 @@ public boolean insertRecord(
       required = recordSize + UAO_SIZE;
     }
 
-    if (getUsed() > pushSortMemoryThreshold
+    long threshold =
```

Comment:
We can set the actual threshold when `growThresholdIfNeeded()` is called and returns true instead of checking for each insertion.